### ISSUE TITLE
DataLoader worker (pid 13904) is killed by signal: Killed. 

### ISSUE THREAD

**Issue by GrowthBreakthru** [guanyingc/SDPS-Net#16]
While training the First stage: train Lighting Calibration Network (LCNet) I got the following error:

File "/home/upt/anaconda3/envs/PST_Environment_1/lib/python3.7/site-packages/torch/utils/data/_utils/signal_handling.py", line 66, in handler
    _error_if_any_worker_fails()
RuntimeError: DataLoader worker (pid 13904) is killed by signal: Killed. 

I am running currently without CUDA. I have 4 CPU cores. 

I have looked online at this problem and have found quite a bit of discussion on this type of error, but I haven't found a solid fix yet. Most of the discussion involves memory usage and the number of workers. Someone suggested increasing the swap file size. I checked the size with the command "swapon --show". Here's what I got: 

NAME      TYPE      SIZE USED PRIO
/dev/dm-0 partition   4G   1G      -2

I tried changing the swap file size with the following commands I found online, but it didn't work. I got an error.

fallocate -l 32G /tmp/swap
chmod 600 /tmp/swap
mkswap /tmp/swap
swapon /tmp/swap

If you have any suggestions that would be great! The training ran a long time before I got the error so this problem could take me a real long time to troubleshoot if I don't have good direction on what to try.

**Comment by GrowthBreakthru** [comment]
There was a lot of good discussion about this problem in this forum discussion: 

https://forums.fast.ai/t/runtimeerror-dataloader-worker-is-killed-by-signal/31277/40

**Comment by guanyingc** [comment]
Hi, are you using CPU for training? I am sorry that I have not tried using CPU for training and haven't confronted this kind of error before. Not sure if using a smaller number of worker helps (--workers)?

**Comment by GrowthBreakthru** [comment]
Thank you for your reply!

I am using CPU for training. I changed the number of workers from 8 to 2
and the batch size from 32 to 4. I also made another adjustment which I can
post on Monday when I'm back at work.

Now the Stage 1 training has run for over 24 hours without a problem. I am
letting it run over the weekend, and I will check it again on Monday.

One other thing is that I used the htop command and now I can watch the CPU
usage for each core and the memory usage. The memory usage has been staying
low so I am hopeful that the problem is fixed.

Thank you,
John High
